lib/probe_dock_ruby/test_result.rb

```diff
@@ -14,15 +14,44 @@ def initialize project, options = {}
         raise Error, "The :duration options is required (indicates how long it took to run the test)"
       end
 
-      @key = options[:key]
       @fingerprint = options[:fingerprint]
+
       @name = options[:name].to_s
 
-      @category = options[:category] || project.category
-      @tags = (wrap(options[:tags]) + wrap(project.tags)).compact.collect(&:to_s).uniq
-      @tickets = (wrap(options[:tickets]) + wrap(project.tickets)).compact.collect(&:to_s).uniq
+      if @name.match(Annotation::ANNOTATION_REGEXP)
+        name_annotation = Annotation.new(@name)
+        @name = Annotation.strip_annotations(@name)
+      end
+
+      if options[:annotation]
+        if options[:annotation].kind_of?(String)
+          options_annotation = Annotation.new(options[:annotation])
+        else
+          options_annotation = options[:annotation]
+        end
+      end
+
+      if name_annotation && options_annotation
+        annotation = name_annotation.merge(options_annotation)
+      else
+        annotation = name_annotation || options_annotation
+      end
+
+      annotation = Annotation.new('') unless annotation
```

Comment:
If you're going to create an annotation anyway, I would write the blocks above like this:

```
annotation = Annotation.new('')

if @name.match(Annotation::ANNOTATION_REGEXP)
  annotation = annotation.merge(Annotation.new(@name))
  @name = Annotation.strip_annotations(@name)
end

options_annotation = options[:annotation]
options_annotation = Annotation.new(options_annotation) if options_annotation.kind_of?(String)
annotation = annotation.merge(options_annotation) if options_annotation
```